frontend/src/api/apiMergeUtils.ts

```diff
@@ -0,0 +1,8 @@
+import { QueryParams } from '@openshift/dynamic-plugin-sdk-utils';
+import { K8sAPIOptions } from 'k8sTypes';
+
+export const mergeK8sQueryParams = (opts: K8sAPIOptions = {}): QueryParams => {
```

Comment:
```suggestion
export const mergeK8sQueryParams = (opts: K8sAPIOptions = {}, specificOpts?: QueryParams = {}): QueryParams => {
```
Allow for additional options that we are not supporting directly to just be merged in.

```
{
  ...specificOpts,
  ...(opts.dryRun && { dryRun: 'All' }),
}
```

If you search for `queryOptions:` you can see we use `labelSelector` right now and I'd like this function to merge any manual operation that is exposed in the type proper to be something we can just add it in if we want -- but for any unknown values the `[key: string]: string` part, we have our type for.

```
// eg. this is the type from the SDK
declare type QueryParams = Partial<{
    watch: string;
    labelSelector: string;
    fieldSelector: string;
    resourceVersion: string;
    [key: string]: string;
}>;
```

Reply:
should `opts` type just be combined with QueryParams instead of having two input params for the merge util.

```ts
export type K8sAPIOptions = QueryParams & {
  dryRun?: boolean;
};
```

Reply:
No -- keep it simple for us with just custom `dryRun`.

---

The why... Simply put, I prefer the model of simplicity calling over exposed internals.

Eg:
```
export const getDSGProjects = (): Promise<ProjectKind[]> => {
  return getProjects('opendatahub.io/dashboard=true');
};
```

Was this function necessary? It's just got a static label. We could just call `getProjects` with what we want at source. It's only used once.

I think this is a superior internal function definition because you are being declarative with what you want. "I want all dsg projects" instead of "I want all projects, here is the internal label to filter them by" (even if we had that as a constant) -- it puts the onus on the caller to know more internals and it bleeds layers of the app together.

This is the same reason I don't really want to see a bunch of API calls like:
```
getK8sResource({ model, queryOptions: { name, ns, queryParams: { labelSelector, dryRun } } })
```
We could just use the SDK everywhere instead of this extra step -- I personally believe having this layer helps with usage, understanding, and isolates functionality away from Modals, Forms, etc ... code that needs to talk more about interactions with the DOM and less about how to talk to the API.

Things like `labelSelector` are internal logic typically, not for the UI component to concern itself with -- this is the separation I would like to see.

@Gkrumbach07 let me know if this was too all over the place or you have further questions. Happy to talk about this more if you want.

Reply:
Is this what you ment?

```ts
export const mergeK8sQueryParams = (
  opts: K8sAPIOptions = {},
  specificOpts: QueryParams = {},
): QueryParams => {
  return {
    ...specificOpts,
    ...(opts.dryRun && { dryRun: 'All' }),
  };
};
```

Reply:
or should specific opts be 
```ts
specificOpts: {[key: string]: string} = {}
```